src/commands/domain/get.js

```diff
@@ -0,0 +1,63 @@
+const { flags } = require('@oclif/command')
+const { getDomainIdentifierArg, reqType, resolvedParam, splitPathParams } = require('../../support/command/parse-input')
+const { from, hasJsonStructure, prettyPrintJSON } = require('../../utils/general')
+const { getDomain } = require('../../requests/domain')
+const { getResponseContent } = require('../../support/command/handle-response')
+const BaseCommand = require('../../support/command/base-command')
+
+class GetDomainCommand extends BaseCommand {
+  constructor(...props) {
+    super(...props)
+    this.logDefinition = response => {
```

Comment:
The same here, we can remove the constructor.

Reply:
I don't follow what you mean. The `logDefinition` method is defined in the constructor. As mentioned above, your suggestion didn't work with node.js 10.

Reply:
yeah, in node you have to do:
```
class GetDomainCommand extends BaseCommand {
    logDefinition(response) {
        const definition = getResponseContent(response)
        this.log(hasJsonStructure(definition)
            ? prettyPrintJSON(definition)
            : definition)
    }

    async run() {
        ...
    }
}
```

Reply:
`TypeError: Cannot read property 'log' of undefined`
i.e. we need "this" to reference the Command object where `log()` is defined.

Reply:
Ah! Nice, then we have to bind it in the constructor as we're throwing `this` down the chain. The original piece is good!

This is why OOP is lame! Functional 4 lyfe!

Reply:
Yes, I was looking for a better pattern here but it's not working with the ES6 arrow function.

Reply:
So going back to the original style is better now.

Reply:
Thanks for the suggestions, I'll revert back to the original style. The `Command` is part of oclif - written in TypeScript so that's the one area that we're using OOP.